


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: CustomErrorController</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">com.taltech.foodapp.controller</a> ]
</div>

<h1>Coverage Summary for Class: CustomErrorController (com.taltech.foodapp.controller)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">CustomErrorController</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    20%
  </span>
  <span class="absValue">
    (1/ 5)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    5.9%
  </span>
  <span class="absValue">
    (1/ 17)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;package com.taltech.foodapp.controller;
<i>2</i>&nbsp;
<i>3</i>&nbsp;import com.taltech.foodapp.dto.ErrorResponseObj;
<i>4</i>&nbsp;import com.taltech.foodapp.exceptions.AuthenticationException;
<i>5</i>&nbsp;import com.taltech.foodapp.exceptions.CategoryException;
<i>6</i>&nbsp;import com.taltech.foodapp.exceptions.ProductException;
<i>7</i>&nbsp;import com.taltech.foodapp.exceptions.UserException;
<i>8</i>&nbsp;import org.springframework.http.HttpStatus;
<i>9</i>&nbsp;import org.springframework.http.ResponseEntity;
<i>10</i>&nbsp;import org.springframework.web.bind.annotation.ControllerAdvice;
<i>11</i>&nbsp;import org.springframework.web.bind.annotation.ExceptionHandler;
<i>12</i>&nbsp;
<i>13</i>&nbsp;import java.util.HashMap;
<i>14</i>&nbsp;import java.util.Map;
<i>15</i>&nbsp;
<i>16</i>&nbsp;@ControllerAdvice
<b class="fc"><i>17</i>&nbsp;class CustomErrorController{</b>
<i>18</i>&nbsp;
<i>19</i>&nbsp;    @ExceptionHandler(value = AuthenticationException.class)
<i>20</i>&nbsp;    public ResponseEntity&lt;Map&lt;String, ErrorResponseObj&gt;&gt; authException(AuthenticationException ex){
<b class="nc"><i>21</i>&nbsp;        ErrorResponseObj errorResponse= new ErrorResponseObj(ex.getCode(), HttpStatus.valueOf(ex.getStatus()), ex.getMessage(), ex.getField());</b>
<b class="nc"><i>22</i>&nbsp;        Map&lt;String, ErrorResponseObj&gt; error= new HashMap&lt;&gt;();</b>
<b class="nc"><i>23</i>&nbsp;        error.put(&quot;error&quot;, errorResponse);</b>
<b class="nc"><i>24</i>&nbsp;        return ResponseEntity.status(errorResponse.getStatus()).body(error);</b>
<i>25</i>&nbsp;    }
<i>26</i>&nbsp;
<i>27</i>&nbsp;    @ExceptionHandler(value = UserException.class)
<i>28</i>&nbsp;    public ResponseEntity&lt;Map&lt;String, ErrorResponseObj&gt;&gt; userException(UserException ex){
<b class="nc"><i>29</i>&nbsp;        ErrorResponseObj errorResponse= new ErrorResponseObj(ex.getCode(), HttpStatus.valueOf(ex.getStatus()), ex.getMessage(), ex.getField());</b>
<b class="nc"><i>30</i>&nbsp;        Map&lt;String, ErrorResponseObj&gt; error= new HashMap&lt;&gt;();</b>
<b class="nc"><i>31</i>&nbsp;        error.put(&quot;error&quot;, errorResponse);</b>
<b class="nc"><i>32</i>&nbsp;        return ResponseEntity.status(errorResponse.getStatus()).body(error);</b>
<i>33</i>&nbsp;    }
<i>34</i>&nbsp;
<i>35</i>&nbsp;    @ExceptionHandler(value = CategoryException.class)
<i>36</i>&nbsp;    public ResponseEntity&lt;Map&lt;String, ErrorResponseObj&gt;&gt; catException(CategoryException ex){
<b class="nc"><i>37</i>&nbsp;        ErrorResponseObj errorResponse= new ErrorResponseObj(ex.getCode(), HttpStatus.valueOf(ex.getStatus()), ex.getMessage(), ex.getField());</b>
<b class="nc"><i>38</i>&nbsp;        Map&lt;String, ErrorResponseObj&gt; error= new HashMap&lt;&gt;();</b>
<b class="nc"><i>39</i>&nbsp;        error.put(&quot;error&quot;, errorResponse);</b>
<b class="nc"><i>40</i>&nbsp;        return ResponseEntity.status(errorResponse.getStatus()).body(error);</b>
<i>41</i>&nbsp;    }
<i>42</i>&nbsp;
<i>43</i>&nbsp;    @ExceptionHandler(value = ProductException.class)
<i>44</i>&nbsp;    public ResponseEntity&lt;Map&lt;String, ErrorResponseObj&gt;&gt; prodException(ProductException ex){
<b class="nc"><i>45</i>&nbsp;        ErrorResponseObj errorResponse= new ErrorResponseObj(ex.getCode(), HttpStatus.valueOf(ex.getStatus()), ex.getMessage(), ex.getField());</b>
<b class="nc"><i>46</i>&nbsp;        Map&lt;String, ErrorResponseObj&gt; error= new HashMap&lt;&gt;();</b>
<b class="nc"><i>47</i>&nbsp;        error.put(&quot;error&quot;, errorResponse);</b>
<b class="nc"><i>48</i>&nbsp;        return ResponseEntity.status(errorResponse.getStatus()).body(error);</b>
<i>49</i>&nbsp;    }
<i>50</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2019-10-27 13:00</div>
</div>
</body>
</html>
